File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& arr) {
        
        int l = 0;
        int r  = arr.size()-1;
        int ans=0;
        while(l<r){
            
            ans = max(ans,min(arr[l],arr[r])*(r-l));
            
            if(arr[l]<arr[r]){
                l++;
            }else{
                r--;
            }
            
            
        }
        
        return ans;
        
    }
};

File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, arr: List[int]) -> int:
        
        l = 0
        r = len(arr)-1
        
        ans =0
        
        while(l<r):
            
            ans = max(ans,min(arr[l],arr[r])*(r-l))
            
            if(arr[l]<arr[r]):
                l+=1
            else:
                r-=1
        
        
        return ans
        